packages/foam-vscode/src/features/hover-provider.ts

```diff
@@ -0,0 +1,81 @@
+import * as vscode from 'vscode';
+import { Foam, FoamWorkspace, ResourceParser, URI, Range } from 'foam-core';
+import { FoamFeature } from '../types';
+import { getNoteTooltip, mdDocSelector, isSome } from '../utils';
+import { toVsCodeRange } from '../utils/vsc-utils';
+import { ResourceLink } from 'foam-core';
+import { getFoamVsCodeConfig } from '../services/config';
+
+const feature: FoamFeature = {
+  activate: async (
+    context: vscode.ExtensionContext,
+    foamPromise: Promise<Foam>
+  ) => {
+    if (!getFoamVsCodeConfig('links.hover.enable')) {
```

Comment:
if we put this check here, changes in the configuration won't be reflected in the editor (the user would have to reload the editor).

I can see two options here:
- do the check in the `provideHover` function
- use `monitorFoamVsCodeConfig` and pass the function (see `document-decorator.ts`)

The first approach is more straightforward but I don't love the "singleton magic".
The second approach has more explicit dependency injection and is the pattern I am currently favoring (although I am open to different arguments).

Let me know your thoughts!

Reply:
I tried to implement the second approach in https://github.com/foambubble/foam/pull/728/commits/4f407f186ba0661e29f098c771318c1567cbf85b What do you think ?

For the first approach, I'm concerned about reading the configuration each time.